assets/js/search.js

```diff
@@ -122,21 +122,37 @@ function titleExtractor (document) {
 
   if (type === 'function' || type === 'callback' || type === 'type') {
     var modFun = title.replace(/\/\d+/, '')
-    var modOrFun = modFun.replace('.', ' ')
+    var modOrFun = modFun.replace(/\./g, ' ')
     var parts = title.split('.')
     title = title + ' ' + modFun + ' ' + modOrFun + ' ' + parts[parts.length - 1]
   }
 
   return title
 }
 
+function snakeCaseSplitter (builder) {
+  function snakeCaseFunction (token) {
+    var snakeTokens =  token.toString().split("_").map(function (str) {
+      return token.clone().update(function () { return str })
```

Comment:
My JavaScript-fu is weak, so what does `clone().update(function() { return str })` does? It feels unnatural but I am sure there is a reason. :D

Reply:
My Javascript is non-existent :) It's from the example they provide of splitting up email addresses (https://lunrjs.com/guides/core_concepts.html). The `token` is an object, not just a string, so I think it's capturing the meta-data of the base token but changing the text associated with it.